### PR TITLE
Bug: reset-password enhancements and bug fix

### PR DESCRIPTION
**Pull request by rhutchison**
Resolved issues:
* models are not loaded.  Regression from #736 (https://github.com/meanjs/mean/pull/736/files#diff-d8679454c2d07935926663de0e9cc9a1L34)
* callback was not passed corresponding user data.
* process exits before mail queue is processed.

Enhancements:
* suppress error messages unless debug mailer option is true.
* from e-mail address comes from config.
* added marker messages to each callback line. (Example: [processed/total] Could not send email for x)
* added status message on-complete to tell how many successful e-mails were sent.

### CODE REVIEW

**Comment by mleanos** [comment]
LGTM.. tested, and confirmed it's sending the emails. 

Thanks @rhutchison

**Comment by lirantal** [comment]
Cool, thanks guys.

**Comment by mleanos** [comment]
@lirantal @rhutchison Slight issue with this script. If there aren't any users in the database, this will hang with no exit. It may seem trivial, but if this script is automated or added to a batch it would reek havoc.

You can add this simple fix after the err check at the beginning of the User.find() callback
```
if (!users.length) {
      console.log();
      console.log(chalk.yellow('No users were found.'));
      process.exit(0);
    }
```
I have a more elegant solution to the *no users found* issue in the PR i submitted to your fork for the user-refactor email migration.

**Comment by rhutchison** [comment]
@mleanos please re-test

**Comment by mleanos** [comment]
@rhutchison I'm going to be away from my computer for most of tonight. I'll test it tomorrow morning.

File: scripts/reset-password.js [hunk 1]
```diff
@@ -1,50 +1,90 @@
...
+          return reportAndExit(processedCount, errorCount);
+        }
+      };
+    };
+
+    // report the processing results and exit
+    function reportAndExit(processedCount, errorCount) {
+      var successCount = processedCount - errorCount;
+
+      console.log();
```
Comment (ilanbiala) [inline scripts/reset-password.js]: Extra?

**Comment by mleanos** [comment]
@rhutchison I've tested this and it works great. However, what's the purpose of the `alerts()`? No alert dialog's popped up for me; is that the intention? If so, I think merely a console log is sufficient.

File: scripts/reset-password.js (hunk 1 above)
Comment (mleanos) [inline scripts/reset-password.js]: @ilanbiala I like the extra line written here. It separates any *line item* messages from the overall processing results.

Comment (rhutchison) [inline scripts/reset-password.js]: @ilanbiala provides line separation, as @mleanos mentioned.  It was intentional

**Comment by rhutchison** [comment]
@mleanos alert is just a mapped function to chalk.green, chalk.red, or chalk.yellow depending on the result status.

**Comment by mleanos** [comment]
@rhutchison thank you. I obviously, didn't pay enough attention. Jumped the gun, when I saw the alert :)

LGTM.

**Comment by lirantal** [comment]
Great. Thanks @rhutchison for fixing this.

**Pull request merged**